src/main/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReference.java

```diff
@@ -170,6 +170,9 @@ public J visitLambda(J.Lambda lambda, ExecutionContext ctx) {
                 if (methodType != null && !isMethodReferenceAmbiguous(methodType)) {
                     if (methodType.hasFlags(Flag.Static) ||
                         methodSelectMatchesFirstLambdaParameter(method, lambda)) {
+                        if (method.getType() instanceof JavaType.Parameterized) {
```

Comment:
@knutwannheden did you mean something like this? Or should we restrict it even further?
```suggestion
                        if (method.getType() instanceof JavaType.Parameterized &&
                            ((JavaType.Parameterized) method.getType()).getTypeParameters().stream()
                                    .anyMatch(JavaType.GenericTypeVariable.class::isInstance)) {
```

Reply:
I was thinking something like that, yes. Otherwise we exclude a lot of cases where the recipe actually works correctly. But I can't say if this is 100% correct or not, as the type inference topic is rather complex. So if we can ensure that the new test case doesn't break and possibly some new test cases assert that it still works for other parameterized return types, then that is good enough, I think.

As an extension we could also check if all of the generic type variables are defined by the method itself. In that case I think the method reference replacement should be safe.